.github/workflows/deploy-to-remote-repository.yml

```diff
@@ -0,0 +1,105 @@
+name: Sync to remote repository
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        default: "ubuntu-latest"
+        required: false
+        type: string
+      remote_repo:
+        description: 'Remote repository URL'
+        required: true
+        type: string
+      remote_branch:
+        description: 'Remote branch to clone and push'
+        required: false
+        type: string
+      base_directory:
+        description: 'Base directory for rsync'
+        required: false
+        default: '.'
+        type: string
+      destination_directory:
+        description: 'Destination directory for rsync in remote repository'
+        required: false
+        default: '.'
+        type: string
+      exclude_list:
+        description: 'Comma-separated list of files and directories to exclude from sync'
+        required: false
+        default: '.git, .gitmodules'
+        type: string
+    secrets:
+      REMOTE_REPO_SSH_KEY:
+        description: 'SSH key for remote repository'
+        required: true
+
+jobs:
+  deploy:
+    runs-on: ${{ inputs.os }}
+    name: "Syncing ${{ inputs.base_directory }} to ${{ inputs.remote_repo }}@${REMOTE_BRANCH}: ${{ inputs.destination_directory }}"
+
+    env:
+      GIT_SSH_COMMAND: ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i ~/.ssh/private_key
+      REMOTE_REPO_DIR: /tmp/remote_repo
```

Comment:
Is there any benefit here of using `${{ inputs.remote_repo }}` to generate the temporary directory? Is it likely to have conflicts?

Considering this project is open source, I mean. Anyone could use it.

Reply:
haha :) I wrote a really long-winded explanation and fat-fingered deleted it... so now you're stuck with the TLDR; 

It's possible it may be of value to someone in the future, but it's difficult to parse from so many possible remote_repo formats and would be better as an override/direct input itself. If we did make it an input variable, there's the potential that someone might misunderstand and clone their remote repo into their current repo's folder structure... asking rsync to recursively sync itself, etc... 

possible to do in the future, but sufficient complexity that leaving it static seems ok for now!